elements/lisk-utils/src/math/q.ts

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable no-bitwise */
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { MAX_FRAC, ONE, Q_OPERATION, TWO } from './constants';
+
+const numberToQ = (base: bigint, val: number): bigint => {
+	const denominator = TWO ** base;
+	const int = Math.floor(val);
+	const decimals = val - int;
+	if (decimals === 0) {
+		return BigInt(val) * denominator;
+	}
+	const [_, binaryStr] = decimals.toString(2).split('.');
```

Comment:
A super tiny remark: variable `_` is not necessary here 🙂

This is a valid JavaScript/TypeScript syntax:

```typescript
const [, binaryStr] = decimals.toString(2).split('.');
```